runtime/pendulum/src/assets.rs

```diff
@@ -0,0 +1,107 @@
+#![allow(non_snake_case)]
+
+pub mod xcm_assets {
+	/// Creates a function and a const u8 representation of the value.
+	/// # Examples
+	/// `create_id!(PARACHAIN_ASSET,100);`
+	///
+	/// will look like:
+	/// ```
+	/// use spacewalk_primitives::CurrencyId;
+	///
+	/// pub const PARACHAIN_ASSET : u8 = 100;
+	/// pub fn PARACHAIN_ASSET_id() -> CurrencyId {
+	/// 	CurrencyId::XCM(PARACHAIN_ASSET)
+	/// }
+	/// ```
+	macro_rules! create_id {
+		($asset_name:ident, $u8_repr:literal) => {
+			paste::item! {
+				pub const $asset_name :u8 = $u8_repr;
+
+				pub fn [< $asset_name _id >]() -> crate::CurrencyId {
+					crate::CurrencyId::XCM($asset_name)
+				}
+			}
+		};
+	}
+
+	create_id!(RELAY_DOT, 0);
+	create_id!(ASSETHUB_USDT, 1);
+	create_id!(ASSETHUB_USDC, 2);
+	create_id!(EQUILIBRIUM_EQD, 3);
+	create_id!(MOONBEAM_BRZ, 4);
+	create_id!(POLKADEX_PDEX, 5);
+}
+
+/// Locations for native currency and all natively issued tokens
+pub mod native_locations {
+	use crate::ParachainInfo;
+	use frame_support::traits::PalletInfoAccess;
+	use xcm::latest::{
+		Junction::{GeneralIndex, PalletInstance, Parachain},
+		Junctions::{X1, X2, X3, X4},
+		MultiLocation,
+	};
+
+	const TOKEN_IN_CURRENCY_ID: u128 = 4;
+
+	fn tokens_pallet_id() -> u8 {
+		crate::Tokens::index() as u8
+	}
+
+	fn balances_pallet_id() -> u8 {
+		crate::Balances::index() as u8
+	}
+
+	/// location of the native currency from the point of view of Pendulum parachain
+	pub fn native_location_local_pov() -> MultiLocation {
+		MultiLocation { parents: 0, interior: X1(PalletInstance(balances_pallet_id())) }
+	}
+
+	/// location of the native currency from the point of view of other parachains(external)
+	pub fn native_location_external_pov() -> MultiLocation {
+		MultiLocation {
+			parents: 1,
+			interior: X2(
+				Parachain(ParachainInfo::parachain_id().into()),
+				PalletInstance(balances_pallet_id()),
+			),
+		}
+	}
```

Comment:
Should we try to make this re-usable somehow so that we can use it on all our chains ie also on Amplitude and Foucoco? It would be nice to have such a definition for all networks but with the current setup, we do this only for Pendulum. Maybe we just copy this `assets.rs` files also to the other two chains and remove all assets that are not used?

Reply:
added assets.rs on foucoco and amplitude; also renamed `create_id!` to  `create_xcm_id!` and moved this to **`runtime-common`**